src/backups2datalad/adataset.py

```diff
@@ -98,15 +97,17 @@ async def ensure_installed(
         return True
 
     async def is_dirty(self) -> bool:
-        return await anyio.to_thread.run_sync(attrgetter("dirty"), self.ds.repo)
-
-    async def is_unclean(self) -> bool:
-        def _unclean() -> bool:
-            return any(
-                r["state"] != "clean" for r in self.ds.status(result_renderer=None)
+        return (
+            await self.read_git(
+                "status",
+                "--porcelain",
+                # Forcibly use default values for these options in case they
+                # were overridden by user's gitconfig:
+                "--untracked-files=normal",
+                "--ignore-submodules=none",
             )
-
-        return await anyio.to_thread.run_sync(_unclean)
+            != ""
+        )
```

Comment:
I think it would indeed be more efficient but not yet "most efficient" for the case of having heavy changes: would it result in early termination of that `git status --porcelain ...` on receiving any output, which already would be indicative of being dirty?

Reply:
No.  I'm not sure if changing the code to terminating `git status` early would result in `git` cleaning everything up properly.

Reply:
I bet it should be fine since I believe it is typically used as `git status --porcelain --untracked-files=normal --ignore-submodules=none | grep -q .` which results in similar early closure of the pipe etc.

But anyways -- probably most of our cases are not that heavy on the diff here so would not really be anyhow notably different. Let's proceed.